Sources/NIOSingletonResources/SingletonType.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if canImport(Darwin)
+import Darwin
+#elseif os(Windows)
+import ucrt
+import WinSDK
+#elseif canImport(Glibc)
+import Glibc
+#elseif canImport(Musl)
+import Musl
+#else
+#error("The concurrency NIOLock module was unable to identify your C library.")
+#endif
+import NIOCore
+
+/// Singleton resources provided by SwiftNIO for programs & libraries that don't need full control over all operating system resources.
+///
+/// SwiftNIO allows and encourages the precise management of all operating system resources such as threads and file descriptors.
+/// Certain resources (such as the main `EventLoopGroup`) however are usually globally shared across the program. This means
+/// that many programs have to carry around an `EventLoopGroup` despite the fact the don't require the ability to fully return
+/// all the operating resources which would imply shutting down the `EventLoopGroup`. This type is the global handle for singleton
+/// resources that applications (and some libraries) can use to obtain never-shut-down singleton resources.
+///
+/// Programs and libraries that do not use these singletons will not incur extra resource usage, these resources are lazily initialized on
+/// first use.
+public enum NIOSingletons {}
+
+extension NIOSingletons {
+    /// How many threads will be used for the multi-threaded `EventLoopGroup`s.
+    ///
+    /// Uses `NIO_SINGLETONS_MULTI_THREADED_ELG_THREAD_COUNT`.
+    public static var suggestedMultiThreadedEventLoopGroupThreadCount: Int {
```

Comment:
It feels odd having this defined in `NIOSingletonResources` when MTELG is defined in `NIOPosix` -- is this used elsewhere too (e.g. for NIOTS)? If so we should consider a less specific name.

Reply:
Yes correct, that's used in NIOTS too and is suggested to be used in all multi-thread/queue/...'ed ELGs.

Reply:
c.f. https://github.com/apple/swift-nio-transport-services/pull/180/files#diff-d899aed582799046dcdaf6578f5f68ec07f205961e912528abb567f3413a4191R42

Reply:
Cool, can we name it something less MTELG-specific then? Maybe `suggestedEventLoopGroupSize`/`suggestedEventLoopGroupLoopCount`?

Reply:
`loopCount` is great! "size" is a little unclear I think

Reply:
fixed